File: 프로그래머스/3/43238. 입국심사/입국심사.java
import java.util.*;

class Solution {
    public long solution(int n, int[] times) {
        Arrays.sort(times);
        long answer = 0;
        long low = 0;
        long high = times[times.length - 1] * (long) n;
        long mid = 0;
        
        while(low <= high) {
            mid = (low + high) / 2;
            long sum = 0;
            
            for(int i=0; i<times.length; i++) {
                sum += mid / times[i];
            }
            
            if(sum < n) {
                low = mid + 1;
            } else {
                high = mid - 1;
                answer = mid;
            }
        }
        return answer;
    }
}

File: 프로그래머스/3/43238. 입국심사/입국심사.kt
class Solution {
    fun solution(n: Int, times: IntArray): Long {
        var answer: Long = 0
        var low: Long = 0
        var high = times[times.size - 1] * n.toLong()
        var mid: Long = 0

        while (low <= high) {
            mid = (low + high) / 2
            var sum: Long = 0

            for (i in times.indices) {
                sum += mid / times[i]
            }

            if (sum < n) {
                low = mid + 1
            } else {
                high = mid - 1
                answer = mid
            }
        }
        times.sort()

        return answer
    }
}

fun main() {
    val result = Solution().solution(
        n = 6,
        times = intArrayOf(7, 10),
    )
    println(result)
}